src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -3354,33 +3396,63 @@ public class PlayerSpigotMock extends Player.Spigot
 
 		@Override
 		@Deprecated
-		public void sendMessage(@NotNull BaseComponent @NotNull ... components)
+		public void sendMessage(@NotNull BaseComponent component)
 		{
-			sendMessage(ChatMessageType.CHAT, components);
+			this.sendMessage(ChatMessageType.SYSTEM, component);
 		}
 
 		@Override
 		@Deprecated
-		public void sendMessage(@NotNull ChatMessageType position, @NotNull BaseComponent @NotNull ... components)
+		public void sendMessage(@NotNull BaseComponent... components)
 		{
-			Preconditions.checkNotNull(position, "Position must not be null");
-			Preconditions.checkNotNull(components, "Component must not be null");
-			Component comp = BungeeComponentSerializer.get().deserialize(components);
-			PlayerMock.this.sendMessage(comp);
+			this.sendMessage(ChatMessageType.SYSTEM, components);
 		}
 
 		@Override
 		@Deprecated
-		public void sendMessage(@NotNull BaseComponent component)
+		public void sendMessage(@Nullable UUID sender, @NotNull BaseComponent component)
 		{
-			sendMessage(ChatMessageType.CHAT, component);
+			this.sendMessage(ChatMessageType.CHAT, sender, component);
+		}
+
+		@Override
+		@Deprecated
+		public void sendMessage(@Nullable UUID sender, @NotNull BaseComponent... components)
+		{
+			this.sendMessage(ChatMessageType.CHAT, sender, components);
 		}
 
 		@Override
 		@Deprecated
 		public void sendMessage(@NotNull ChatMessageType position, @NotNull BaseComponent component)
 		{
-			sendMessage(position, new BaseComponent[]{ component });
+			this.sendMessage(position, new BaseComponent[]{ component });
+		}
+
+		@Override
+		@Deprecated
+		public void sendMessage(@NotNull ChatMessageType position, @NotNull BaseComponent... components)
+		{
+			this.sendMessage(position, null, components);
+		}
+
+		@Override
+		@Deprecated
+		public void sendMessage(@NotNull ChatMessageType position, @Nullable UUID sender, @NotNull BaseComponent component)
+		{
+			this.sendMessage( position, sender, new BaseComponent[] { component } );
+		}
+
+		@Override
+		@Deprecated
+		public void sendMessage(@NotNull ChatMessageType position, @Nullable UUID sender, @NotNull BaseComponent... components)
+		{
+			Preconditions.checkNotNull(position, "Position must not be null");
+			Preconditions.checkNotNull(components, "Component must not be null");
+			Component comp = BungeeComponentSerializer.get().deserialize(components);
+			String serialized = LegacyComponentSerializer.legacySection().serialize(comp);
+			comp = LegacyComponentSerializer.legacySection().deserialize(serialized);
```

Comment:
This is the fix for the issue mentioned in the description.
After deserialize using Bungeecord API we convert it to a String and back to a component to match the expected behaviour.